src/pal/src/map/map.cpp

```diff
@@ -2651,3 +2651,60 @@ BOOL MAPUnmapPEFile(LPCVOID lpAddress)
     TRACE_(LOADER)("MAPUnmapPEFile returning %d\n", retval);
     return retval;
 }
+
+/*++
+Function :
+    MAPEraseSection - erase a section
+    returns TRUE if successful, FALSE otherwise
+--*/
+BOOL MAPEraseSection(LPCVOID lpAddress)
+{
+    TRACE_(LOADER)("MAPEraseSection(lpAddress=%p)\n", lpAddress);
+
+    if ( NULL == lpAddress )
+    {
+        ERROR_(LOADER)( "lpAddress cannot be NULL\n" );
+        return FALSE;
+    }
+
+    BOOL retval = FALSE;
+    CPalThread * pThread = InternalGetCurrentThread();
+    InternalEnterCriticalSection(pThread, &mapping_critsec);
+    PLIST_ENTRY pLink, pLinkNext = NULL;
+
+    // Look through the entire MappedViewList for all mappings associated with the
+    // section with an address 'lpAddress'. We want to erase.
+
+    for(pLink = MappedViewList.Flink;
+            pLink != &MappedViewList;
+            pLink = pLinkNext)
+    {
+        pLinkNext = pLink->Flink;
+        PMAPPED_VIEW_LIST pView = CONTAINING_RECORD(pLink, MAPPED_VIEW_LIST, Link);
+
+        if (pView->lpAddress == lpAddress) // this entry is associated with the section
+        {
+            if (-1 == munmap(pView->lpAddress,
+                             pView->NumberOfBytesToMap))
+            {
+                ERROR_(LOADER)("Unable to unmap the section for erasing. Expect trouble.\n");
+            } else if (MAP_FAILED == mmap(pView->lpAddress,
```

Comment:
I am wondering whether we can just leave the reloc section unmapped.

Reply:
I found that other dll file can be mapped to unmapped memory.
CoreCLR fails to run because the range of dll files are overlapped.
(ex. _ASSERTE(pnewrange->LowAddress >= current->HighAddress); in  ExecutionManager::AddRangeHelper)